packages/contentful-apps/redirects/src/components/Form.tsx

```diff
@@ -0,0 +1,84 @@
+import { Form as CForm } from '@contentful/f36-components'
+import {
+  ChangeEvent,
+  MutableRefObject,
+  ReactElement,
+  Ref,
+  useImperativeHandle,
+  useRef,
+  useState,
+} from 'react'
+import {
+  DEFAULT_FORM_ERRORS,
+  DEFAULT_FORM_VALUES,
+  FROM_REGEX,
+  TO_REGEX,
+} from '../constants'
+import { Redirect } from '../types'
+import StatusFormControl from './StatusFormControl'
+import UrlFormControl from './UrlFormControl'
+
+type FormProps = {
+  formRef: Ref<any>
+  redirect?: Redirect
+}
+
+const Form = ({ formRef, redirect }: FormProps): ReactElement => {
+  const [formData, setFormData] = useState(redirect || DEFAULT_FORM_VALUES)
+  const [errors, setErrors] = useState(DEFAULT_FORM_ERRORS)
+
+  const fromRef = useRef() as MutableRefObject<HTMLInputElement>
+  const toRef = useRef() as MutableRefObject<HTMLInputElement>
+
+  useImperativeHandle(formRef, () => ({
+    getFormData: (): Redirect => formData,
+  }))
+
+  /**
+   * Tests FROM/TO inputs for errors
+   * @param field
+   * @param value
+   * @returns error message or empty
+   */
+  const getError = (field: 'FROM' | 'TO', value: string): string => {
+    const regex = field === 'FROM' ? FROM_REGEX : TO_REGEX
+    const hasError = !regex.test(value)
+    return hasError ? 'URL provided is not valid' : ''
```

Comment:
I would suggest to create two separate functions `validateFrom` and `validateTo`, in order to have different Error messages.
For the "From" field "URL" is not really correct, since it is rather something like a "path pattern".